mantis-control-plane/mantis-control-plane-client/src/main/java/io/mantisrx/server/master/resourcecluster/ResourceClusterGatewayClient.java

```diff
@@ -128,6 +128,7 @@ private AsyncHttpClient buildCloseableHttpClient(CoreConfiguration configuration
             .setConnectTimeout(configuration.getAsyncHttpClientConnectionTimeoutMs())
             .setRequestTimeout(configuration.getAsyncHttpClientRequestTimeoutMs())
             .setReadTimeout(configuration.getAsyncHttpClientReadTimeoutMs())
+            .setFollowRedirect(configuration.getAsyncHttpClientFollowRedirect())
```

Comment:
just curious how does redirect work for you in this case? This communication is from control plane to an agent which should not get a new IP in general.

Reply:
Agreed. @timmartin-stripe I think we want to catch the location in which communication is coming in the opposite direction. From agent to master is where we run into real issues.